include/vast/Util/Warnings.hpp

```diff
@@ -47,16 +47,54 @@ VAST_RELAX_WARNINGS
 #include <llvm/Support/Debug.h>
 VAST_UNRELAX_WARNINGS
 
+#include <stdexcept>
+
 #define DEBUG_TYPE "vast"
 
+#ifdef ENABLE_VAST_EXCEPTION
+  #define vast_error vast::error<std::runtime_error>
+#else
+  #define vast_error llvm::dbgs
+#endif
+
+#define vast_debug llvm::dbgs
+
 namespace vast
 {
+#ifdef ENABLE_VAST_EXCEPTION
+    template <typename T = std::runtime_error>
+    struct error {
+      private:
+        std::string msg;
+        bool triggered = false;
+
+       public:
+        error(bool cond = true, std::string msg_ = std::string())
+              : msg(std::move(msg_)), triggered(cond) {}
```

Comment:
Argument can be just named `msg` (while I know some people like the codestyle of adding the `_` to the ctor VAST does not do it).